### ISSUE TITLE
Multi-GPU training not giving good accuracy

### ISSUE THREAD

**Issue by sachins9800** [NVIDIA/NeMo#1227]
Hi all,
We experimented with training ASR jasper model using single GPU and multiple GPU's. We noticed the following - 

1) Single GPU training reached a WER of 30% after 50 epochs and WER is still improving significantly with every epoch
2) Multi gpu training only reached a WER of 70% after 50 epochs and WER is stagnated at that number

Can somebody help us get comparable results in both ?. Do we need to pick better hyperparameter values for multi GPU training ?

**Comment by borisgin** [comment]
yes, you will need different learning rate for multi-gpu case since learning rate depends on global batch size=num_gpus * batch_gpu.

**Comment by okuchaiev** [comment]
(Scenario A) Suppose you train on 1 GPU with batch size of X.
(Scenario B) If you leave all other hyper-parameters unchanged but set batch size = X/N and train on N GPUs you should get results comparable to Scenario A but faster. Because total_batch_size = batch_size * number_of_gpus_per_node * number_of_nodes

However, Scenario B would not be the most efficient use of GPUs. 

What we typically do is maximize batch size per GPU and if we train on more GPUs => total batch size is larger => other hyperparams such as learning rate should change. 
If your learning rate is L for 1 GPU, when training on N GPUs, try learning_rate=L*SQRT(N) (or smaller) with a bit longer warmup period.
This is just a first thing to try, you may need to experiment more to get best results
